packages/@ourworldindata/grapher/src/slopeCharts/SlopeChart.tsx

```diff
@@ -132,8 +164,9 @@ export class SlopeChart
         this.selectionArray.toggleSelection(hoverKey)
     }
 
-    @action.bound onLegendMouseOver(color: string) {
-        this.hoverColor = color
+    @action.bound onLegendMouseOver(binOrColor: string | ColorScaleBin) {
```

Comment:
This is unfortunate. Both legend managers accept a `onLegendMouseOver` prop, but the functions have different signatures. The horizontal legend expects a `string`, the vertical legend expects a `ColorScaleBin`.

In general, the manager pattern made it more difficult than it should be to render a second legend component. I think the pass-the-parent-as-manager pattern works well on the top level (i.e. Grapher passes itself to chart components), but for "building block" components like an axis, the manager pattern might not be the best choice.

Reply:
Could you please add a code comment explaining this?

Reply:
👍🏻